### ISSUE TITLE
stacked bar charts

### ISSUE THREAD

**Issue by osilander** [cxli233/FriendsDontLetFriends#6]
An explanation as to why these incredibly popular stacked bar charts should actually be heat maps?

<img width="1018" alt="Screenshot 2023-11-17 at 10 26 56 AM" src="https://github.com/cxli233/FriendsDontLetFriends/assets/25470154/5d5c7368-d2bc-4b39-8636-a5a8632244cc">

**Comment by cxli233** [comment]
Sorry, I don't quite understand this question. Can you elaborate?

**Comment by osilander** [comment]
Sorry, bad explanation. While stacked bar charts can be good, I think they can be very hard to interpret, especially in cases where there are many categories or many bars. (I think it can also be hard to follow the relative heights of different categories across bars).
Regardless, the picture I added shows a _very standard_ plot from a microbiome study. These plots are difficult to interpret, not least because scanning back and forth between the legend and the plot is difficult. If they were presented as heat maps, not only would they be easier to read (taxa in rows, "treatment" in columns (as many gene expression data sets are)), they could be reordered so that similar samples were grouped. This is not possible (or, not performed) for the stacked bar charts -- despite these being the default plot for microbiome studies. In addition, there would be no need to scan back and forth from the legend to the plot. In _addition_, the heatmap scale could be log transformed, for example to visualise low-abundance taxa. Or it could be normalised in some way (e.g. row normalised).
I would love to see more of these microbiome stacked bar charts as heatmaps, and think it might even bring new insight to a lot of previous studies.
Anyway, really nice set of other examples you put up. Cheers.

**Comment by cxli233** [comment]
I think in stacked bar plots, the order of the stacks, as well as the order of the bars, need to be optimized, much like how we reorder rows and columns of a heatmap. 

![image](https://github.com/cxli233/FriendsDontLetFriends/assets/47674556/26db4ccb-7ab0-41fb-af98-d1d7e8d0e1d4)
For example, in this stacked bar plot used to represent population structure of carrots, the stacks and bars are reordered in a way that there is a clear pattern.  Maybe I will do a "friends don't let friends forget to reorder stacked bars" when I get the time. 

Thanks for the comment.
